web/src/containers/LoginPageRoutes.test.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { renderWithConnection } from 'apd-testing-library';
+import LoginPageRoutes from './LoginPageRoutes';
+
+const props = {
+  hasEverLoggedOn: false,
+  fetching: false,
+  factorsList: [
+    {
+      id: 'emf5n20o64XfgZV7E297',
+      factorType: 'email',
+      provider: 'OKTA',
+      vendorName: 'OKTA',
+      profile: {
+        email: 'test@email.com'
+      }
+    }
+  ],
+  verifyData: {
+    qrcode: { href: 'www.test.com' }
+  },
+  handleFactorSelection: jest.fn(),
+  handlePhoneSubmit: jest.fn(),
+  handleVerificationCode: jest.fn(),
+  handleCreateAccessRequest: jest.fn(),
+  handleCompleteAccessRequest: jest.fn(),
+  handleLogin: jest.fn(),
+  handleLoginOtp: jest.fn(),
+  handleLogout: jest.fn()
+};
+
+describe('LoginPageRoutes', () => {
+  it('should redirect the user to Login if the path is /login', () => {
+    const useRouteMatch = jest.fn().mockReturnValue({ path: '/login' });
+    const { queryByRole } = renderWithConnection(
+      <LoginPageRoutes {...props} useRouteMatch={useRouteMatch} />,
+      {
+        initialHistory: ['/login']
+      }
+    );
+    expect(queryByRole('heading', { name: /Log in/i })).toBeTruthy();
+  });
+
+  it('should redirect the user to Account Locked if that path is /login/locked-out', () => {
+    const useRouteMatch = jest.fn().mockReturnValue({ path: '/login' });
+    const { queryByRole } = renderWithConnection(
+      <LoginPageRoutes {...props} useRouteMatch={useRouteMatch} />,
+      {
+        initialHistory: ['/login/locked-out']
+      }
+    );
```

Comment:
Just a visual nitpic: You could define `initialHistory` first, and avoid spreading the `renderOptions` object over three lines:

```
const initialHistory = ['/login/locked-out'];
const {
  queryByRole
} = renderWithConnection(
  <LoginPageRoutes {...props} useRouteMatch={useRouteMatch} />, { initialHistory }
);
```

Prettier might possibly wreck this formatting though. So, just a thought. :man_shrugging:

Reply:
Yeah I was looking for an example of that better component set up that you've been doing, but I couldn't find one.